### PR TITLE
Remove the call that makes miri fail

### PR DESCRIPTION
**Pull request by DutchGhost**
Fixes the concern raised in https://github.com/rust-lang/rust/pull/69645/files#r392884274

cc @RalfJung

### CODE REVIEW

File: src/libcore/tests/mem.rs [hunk 1]
```diff
@@ -142,8 +142,4 @@ fn test_const_forget() {
     const fn const_forget_box<T>(x: Box<T>) {
```
Comment (RalfJung) [inline src/libcore/tests/mem.rs]: Won't this warn due to an unused function?

File: src/libcore/tests/mem.rs [hunk 2]
```diff
@@ -134,12 +134,4 @@ fn test_discriminant_send_sync() {
 fn test_const_forget() {
     const _: () = forget(0i32);
     const _: () = forget(Vec::<Vec<Box<i32>>>::new());
-
-    // Writing this function signature without const-forget
-    // triggers compiler errors:
-    // 1) That we use a non-const fn inside a const fn
-    // 2) without the forget, it complains about the destructor of Box
```
Comment (RalfJung) [inline src/libcore/tests/mem.rs]: I found these quite good reasons to keep the method around. Just add `#[allow(unused)]` to it, and maybe `// FIXME this method cannot be called in const-eval yet`.

**Comment by RalfJung** [comment]
@oli-obk we usually test these things in ui tests, right? Would you prefer this to become a ui test instead?

**Comment by oli-obk** [comment]
Yea a ui test would be more in line with our other tests

**Comment by nikomatsakis** [comment]
I'm confused what's going on here -- but I'll pass the review over to @oli-obk 

r? @oli-obk

**Comment by RalfJung** [comment]
Okay, so @DutchGhost can you make this into a ui test? You can find existing tests in https://github.com/rust-lang/rust/tree/master/src/test/ui/consts.

File: src/test/ui/consts/const_forget.rs [hunk 3]
```diff
@@ -0,0 +1,22 @@
+// run-pass
```
Comment (RalfJung) [inline src/test/ui/consts/const_forget.rs]: Since this is a const-test, why is `check-pass` not sufficient?

Comment (RalfJung) [reply src/test/ui/consts/const_forget.rs]: @DutchGhost this blocks enabling leak checks again in https://github.com/RalfJung/miri-test-libstd, would be good if we could land this quickly. :)

**Comment by RalfJung** [comment]
Thanks, that was quick. :)
@bors r+ rollup

**Pull request merged**